doc/mgmt/Management Framework.md

```diff
@@ -0,0 +1,2029 @@
+    CreateEntry(k,v)|    none          | HMSET(fields in v)| ValidateEditConfig(
+                    |                  |                   |  OP_CREATE)
+    ----------------|------------------|-------------------|--------------------
+    ModEntry(k,v)   | mod_entry(k,v)   | HMSET(fields in v)| ValidateEditConfig(
+                    |                  |                   |  OP_UPDATE)
+    ----------------|------------------|-------------------|--------------------
+    DeleteEntry(k,v)|set,mod_entry(k,0)| DEL               | ValidateEditConfig(
+                    |                  |                   |  OP_DELETE)
+    ----------------|------------------|-------------------|--------------------
+    DeleteEntryField|    none          | HDEL(fields)      | ValidateEditConfig(
+    (k,v)           |                  |                   |   DEL_FIELDS)
+
+
+##### 3.2.2.7 Transformer
+
+Transformer provides the underlying infrastructure for developers to translate data from YANG to ABNF/Redis schema and vice versa, using YANG extensions annotated to YANG paths to provide translation methods. At run time, the YANG extensions are mapped to an in-memory Transformer Spec that provides two-way mapping between YANG and ABNF/Redis schema for Transformer to perform data translation while processing SET/GET operations.
+
+In case that SONiC YANG modules are used by NBI applications, the Transformer performs 1:1 mapping between a YANG object and a SONiC DB object without a need to write special translation codes. If the openconfig YANGs are used by NBI applications, you may need special handling to translate data between YANG and ABNF schema. In such case, you can annotate YANG extensions and write callbacks to perform translations where required.
+
+For special handling, a developer needs to provide:
+1. An annotation file to define YANG extensions on YANG paths where translation required
+e.g. In openconfig-acl.yang, ACL FORWARDING_ACTION "ACCEPT" mapped to "FORWARD", ACL sequence id ‘1’ mapped to ‘RULE_1’ in Redis DB etc. 
+2. Transformer callbacks to perform translation
+
+###### 3.2.2.7.1 Components
+
+Transformer consists of the following components and data:
+* **Transformer Spec:** a collection of translation hints
+* **Spec Builder:** loads YANG and annotation files to dynamically build YANG schema tree and Transformer Spec.
+* **Transformer Core:** perform main transformer tasks, i.e. encode/decode YGOT, traverse the payload, lookup Transformer spec, call Transformer methods, construct the results, error reporting etc.
+* **Built-in Default Transformer method:** perform static translation
+* **Overloaded Transformer methods:** callback functions invoked by Transformer core to perform complex translation with developer supplied translation logic
+* **Ouput framer:** aggregate the translated pieces returned from default and overloaded methods to construct the output payload
+* **Method o
```

Comment:
Does this refer just to the application code or do the generated code fragments also provide syslog messages that we can trace?

Reply:
@daall Method overloader is a part of transformer infra code to override the default transformer and perform data translation with user-supplied callbacks. Yes, the translation processes are syslog-ed for trace.